#0-down,1-up, 3,-max, 4-auto
int32 cmd
---